### PR TITLE
zap2it

### PR DESCRIPTION
**Pull request by th0ma7**
_Motivation:_  Provide a EPG grabber for zap2it package in order to generate XMLTV file output for usage with Jellyfin / Emby / TVheadEnd.  This is also in preparation for DSM7 upgrade where current EPG grabber uses perl and requires manually compiled CPAN modules making it rather complex to maintain.  Current solution is looking into using a python3 option.
_Linked issues:_  https://github.com/daniel-widrick/zap2it-GuideScraping/issues/7

### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

### CODE REVIEW

**Comment by th0ma7** [comment]
@hgy59 any idea how to create a cron job at install time? (e.g. scheduled task I guess from the admin interface).

**Comment by th0ma7** [comment]
cronjob hint... I first started looking into `synoschedtask` but I revisited `synocrond` and found in the binary paths towards system-type cron jobs where there are examples :
```
/usr/local/etc/synocron.d
/usr/syno/etc/synocron.d
/usr/syno/share/synocron.d
```
Where I'm testing the following under `/usr/local/etc/synocron.d` :
```
/usr/local/etc/synocron.d# cat zap2it.conf 
[
   {
      "cmd" : "/usr/local/bin/python3 /var/packages/zap2it/target/bin/zap2it-GuideScrape.py -i /var/packages/zap2it/target/etc/zap2itconfig.ini -o /var/packages/zap2it/target/var/xmlguide.xmltv",
      "crontab" : "0 0,6,12,18 * * *",
      "name" : "zap2itDownloadXMLTVJob",
      "package_username" : "sc-zap2it",
      "period" : "hourly",
      "user" : "package"
   }
]
```

File: spk/zap2it/Makefile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+SPK_ICON = src/zap2it.png
+DSM_UI_DIR = app
+
+DEPENDS = cross/$(SPK_NAME)
+SPK_DEPENDS = "python3>=3.7.7-15"
+
+MAINTAINER = th0ma7
+DESCRIPTION = Takes TV listings from zap2it and convert them to XMLTV for use with applications such as Jellyfin / Emby / TVheadEnd
+DESCRIPTION_FRE = Prend la programmation télé de zap2it et la convertis en XMLTV pour une utilisation avec des applications telles que Jellyfin / Emby / TVheadEnd
+RELOAD_UI = no
```
Comment (hgy59) [inline spk/zap2it/Makefile:13]: RELOAD_UI is obsolete
```suggestion
```

File: spk/zap2it/src/service-setup.sh [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    echo "Set configuration according to the wizard"
+    sed -i "/^Username: /s/ .*/ ${zap2it_user}/" ${ZAP2IT_CONF}
+    sed -i "/^Password: /s/ .*/ ${zap2it_password}/" ${ZAP2IT_CONF}
+    sed -i "/^country: /s/ .*/ ${zap2it_country}/" ${ZAP2IT_CONF}
+    sed -i "/^zipCode: /s/ .*/ ${zap2it_code}/" ${ZAP2IT_CONF}
+    sed -i "/^historicalGuideDays: /s/ .*/ ${zap2it_days}/" ${ZAP2IT_CONF}
+
+    # Encrypt password
+    echo "Encrypt password"
+    ZAP2IT_PASSWD=$(echo -n "${zap2it_password}" | openssl enc -aes-256-cbc -pass pass:zap2it -a)
```
Comment (hgy59) [inline spk/zap2it/src/service-setup.sh:41]: may conflict with already encrypted password (see above: `Keep previous passwd if left blank at upgrade time`)

File: spk/zap2it/src/service-setup.sh [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    sed -i -n -e '/^ZAP2IT_US=/!p' -e "\$aZAP2IT_US=\"${zap2it_US}\"" ${INSTALLER_VARIABLES}
+    sed -i -n -e '/^ZAP2IT_CODE=/!p' -e "\$aZAP2IT_CODE=\"${zap2it_code}\"" ${INSTALLER_VARIABLES}
+    sed -i -n -e '/^ZAP2IT_DAYS=/!p' -e "\$aZAP2IT_DAYS=\"${zap2it_days}\"" ${INSTALLER_VARIABLES}
+}
+
+service_postinst ()
+{
+    # Create cache directory if it does not exist
+    echo "Create cache directory if it does not exist (${ZAP2IT_CACHE})"
+    mkdir -p --mode=0755 ${ZAP2IT_CACHE}
```
Comment (hgy59) [inline spk/zap2it/src/service-setup.sh:55]: I prefer to use different log output whether the cache directory exists or is created here.

File: spk/zap2it/src/service-setup.sh (hunk 2 above)
Comment (th0ma7) [reply spk/zap2it/src/service-setup.sh:41]: it actually decrypt the passwd from to regenerate default configuration file if it was left blank.
Took me a little while to play with this use case.
I was hoping the developer would add a similar feature on his side to remove all blank passwd.

File: spk/zap2it/src/service-setup.sh (hunk 3 above)
Comment (th0ma7) [reply spk/zap2it/src/service-setup.sh:55]: I'm not sure I follow?  Any example to point me to?

Comment (hgy59) [reply spk/zap2it/src/service-setup.sh:55]: Just test whether the cache directory exists and 
- when it does not exist create it and log "cache directory created"
- when it exists log nothing or "cache directory already exists".

Comment (th0ma7) [reply spk/zap2it/src/service-setup.sh]: done

**Comment by th0ma7** [comment]
Closing PR as superseded with https://github.com/SynoCommunity/spksrc/pull/4680

**Pull request closed without merging**